File: classifier/data/raw/ai/samples/project_euler_java/Problem039.java
/** Project Euler 39: perimeter at most 1000 with the most integer right triangles. */
public class Problem039 {
    public static void main(String[] args) {
        int[] solutions = new int[1001];
        for (int a = 1; a <= 1000; a++) {
            for (int b = a; a + b <= 1000; b++) {
                int cSquared = a * a + b * b;
                int c = (int) Math.sqrt(cSquared);
                if (c * c == cSquared && a + b + c <= 1000) {
                    solutions[a + b + c]++;
                }
            }
        }
        int bestPerimeter = 0;
        for (int perimeter = 1; perimeter <= 1000; perimeter++) {
            if (solutions[perimeter] > solutions[bestPerimeter]) {
                bestPerimeter = perimeter;
            }
        }
        System.out.println(bestPerimeter);
    }
}

File: classifier/data/raw/ai/samples/project_euler_zig/problem_039.zig
const std = @import("std");

// Project Euler 39: perimeter at most 1000 with the most integer right triangles.
pub fn main() !void {
    var best_perimeter: u32 = 0;
    var best_count: u32 = 0;
    var perimeter: u32 = 12;
    while (perimeter <= 1000) : (perimeter += 1) {
        var count: u32 = 0;
        var a: u32 = 1;
        while (a < perimeter / 3) : (a += 1) {
            var b = a;
            while (b < (perimeter - a) / 2) : (b += 1) {
                const c = perimeter - a - b;
                if (a * a + b * b == c * c) count += 1;
            }
        }
        if (count > best_count) {
            best_count = count;
            best_perimeter = perimeter;
        }
    }
    try printAnswer(best_perimeter);
}

fn printAnswer(value: anytype) !void {
    var buffer: [64]u8 = undefined;
    const text = try std.fmt.bufPrint(&buffer, "{d}\n", .{value});
    try std.Io.File.stdout().writeStreamingAll(std.Options.debug_io, text);
}
